src/XrdHttp/XrdHttpUtils.cc

```diff
@@ -496,5 +497,140 @@ char *escapeXML(const char *str) {
   return r;
 }
 
+int mapErrNoToHttp(int err) {
```

Comment:
To be complete, let's attempt to map all errors which can be returned by `open`, `read`, `write`, `close`, and `seek` at least. For example, we should add at least the following adjustments:

- `EACCESS` &rarr; 403
- `EPERM` &rarr; 401
- `ENAMETOOLONG` &rarr; 414
- `ETIMEDOUT` &rarr; 408 or 504 (Request timeout or Gateway timeout)

and the following additions:

- `ELOOP` &rarr; 508
- `EFBIG` &rarr; 413 (file contents too big on write)
- `ENXIO` &rarr; 416  (seek for offset outside the file length)
- `EOVERFLOW` &rarr; 406 (seek offset cannot be represented with off_t)